Orange/data/table.py

```diff
@@ -796,17 +796,17 @@ def __setitem__(self, key, value):
             if value is None:
                 value = Unknown
 
-            if not isinstance(value, Real) and \
+            if not isinstance(value, (Real, np.ndarray)) and \
                     (len(attr_cols) or len(class_cols)):
                 raise TypeError(
                     "Ordinary attributes can only have primitive values")
             if len(attr_cols):
-                if len(attr_cols) == 1:
+                if sp.issparse(self.X) and len(attr_cols) == 1:
```

Comment:
Why do we need this? Scipy seems to be able to assign 2d arrays to columns regardless if `attr_cols = 2` or `attr_cols = [2]`. Is this here just for efficiency?
```py
>>> import numpy as np
>>> import scipy.sparse as sp
>>> x = sp.csr_matrix(np.eye(2))
>>> x.toarray()
array([[ 1.,  0.],
       [ 0.,  1.]])
>>> x[:, [1]] = np.array([[2], [2]])
/Users/Niko/anaconda/envs/orange3/lib/python3.5/site-packages/scipy/sparse/compressed.py:774: SparseEfficiencyWarning: Changing the sparsity structure of a csr_matrix is expensive. lil_matrix is more efficient.
  SparseEfficiencyWarning)
>>> x.toarray()
array([[ 1.,  2.],
       [ 0.,  2.]])
>>> x[:, 1] = np.array([[3], [3]])
>>> x.toarray()
array([[ 1.,  3.],
       [ 0.,  3.]])
```